things/service.go

```diff
@@ -549,6 +602,92 @@ func (svc service) Identify(ctx context.Context, key string) (string, error) {
 	return client.ID, nil
 }
 
+func (svc service) VerifyConnections(ctx context.Context, thingIds, groupIds []string) (mgclients.ConnectionsPage, error) {
+	uniqueThings := getUniqueValues(thingIds)
+	uniqueChannels := getUniqueValues(groupIds)
+	totalConnectionsCnt := len(uniqueChannels) * len(uniqueThings)
+	g, ctx := errgroup.WithContext(ctx)
+
+	connections := make([]mgclients.ConnectionStatus, 0, totalConnectionsCnt)
+
+	index := 0
+	for _, th := range uniqueThings {
+		for _, ch := range uniqueChannels {
+			func(thing, channel string, i int) {
+				g.Go(func() error {
+					authReq := &magistrala.AuthorizeReq{
+						Subject:     channel,
+						SubjectType: auth.GroupType,
+						Permission:  auth.GroupRelation,
+						Object:      thing,
+						ObjectType:  auth.ThingType,
+					}
+
+					_, err := svc.auth.Authorize(ctx, authReq)
+					var status mgclients.State
+					switch {
+					case err == nil:
+						status = mgclients.Connected
+					case errors.Contains(err, svcerr.ErrAuthorization):
+						status = mgclients.Disconnected
+					default:
+						return errors.Wrap(svcerr.ErrMalformedEntity, err)
+					}
+
+					connections = append(connections, mgclients.ConnectionStatus{
+						ThingId:   thing,
+						ChannelId: channel,
+						Status:    status,
+					})
+
+					return nil
+				})
+			}(th, ch, index)
```

Comment:
Index is used anywhere ?

Reply:
Its not being used currently. It was used during tests. Let me remove it